extensions/nodes/node_status.go

```diff
@@ -132,6 +133,34 @@ func AllNodeDeleted(client *rancher.Client, ClusterID string) error {
 
 			if len(nodes.Data) == 0 {
 				logrus.Infof("All nodes in the cluster are deleted!")
+
+				return true, nil
+			}
+
+			return false, nil
+		})
+
+	return err
+}
+
+// IsNodeDeleted is a helper method that will loop and check if the node is deleted in the cluster.
+func IsNodeDeleted(client *rancher.Client, nodeName, ClusterID string) error {
+	ctx := context.Background()
+	err := wait.PollUntilContextTimeout(
+		ctx, oneSecondInterval, defaults.ThirtyMinuteTimeout, true, func(ctx context.Context) (bool, error) {
+			node, err := client.Management.Node.List(&types.ListOpts{
```

Comment:
just noting that this is using management. Is there a way to update this to v1 instead? Or is this function only for rke1 (if so could that be noted in the godoc comment)?

Reply:
Same as above.

Reply:
Talked offline, will do this in a separate PR.